netdata/netdata
# Use a separate thread for slow mountpoints in the diskspace plugin

##### Summary
`stat()` and `statvfs()` calls are blocking. That causes gaps in diskspace charts, especially if there are network filesystems mounted to a system. We will check for the time needed to process mountpoints and if the calls are slow, process the "slow" mountpoints in a separate thread with a much longer `update_every`.

Fixes #10150

##### Test Plan
1. Insert `if (!strcmp(mi->mount_point, "/tmp") || !strcmp(mi->mount_point, "/run")) sleep(4);` before or after `stat()`, `statvfs()` calls.
2. Check if `disk_space.` and `disk_inodes.`charts for `/tmp` and `/run` mountpoints work and `update_every` is more than 5 for them.
